src/test/java/com/oath/halodb/SequenceNumberTest.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2018, Oath Inc
+ * Licensed under the terms of the Apache License 2.0. Please refer to accompanying LICENSE file for terms.
+ */
+
+package com.oath.halodb;
+
+import com.google.common.primitives.Longs;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.Matchers;
+import org.testng.annotations.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * @author Pulkit Goel
+ */
+public class SequenceNumberTest extends TestBase {
+
+    @Test(dataProvider = "Options")
+    public void testSequenceNumber(HaloDBOptions options) throws Exception {
+        String directory = TestUtils.getTestDirectory("SequenceNumberTest", "testSequenceNumber");
+
+        int recordSize = 1024;
+        int recordNumber = 100;
+
+        HaloDB db = getTestDB(directory, options);
+
+        byte[] data = new byte[recordSize - Record.Header.HEADER_SIZE - 8 - 8];
```

Comment:
You could use `TestUtils.insertRandomRecords` to generate and insert test data. There are also other utility methods in `TestUtils` which you may find useful.